File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number<=0:
        raise ValueError('Classification is only possible for positive integers.')
    sum=0
    for factor in range(1,number):
        if number%factor==0:
            sum+=factor

    res=''
    if sum==number:
        res='perfect'
    elif sum<number:
        res='deficient'
    else:
        res='abundant'

    return res

File: solutions/dart/perfect-numbers/1/lib/perfect_numbers.dart
// define the Classification enum
enum Classification {
  perfect,
  abundant,
  deficient,
}

class PerfectNumbers {
  Classification classify(int number) {
    if (number <= 0) {
      throw ArgumentError('Number must be a positive integer');
    }

    int sumOfDivisors = _sumOfDivisors(number);

    if (sumOfDivisors == number) {
      return Classification.perfect;
    } else if (sumOfDivisors > number) {
      return Classification.abundant;
    } else {
      return Classification.deficient;
    }
  }

  int _sumOfDivisors(int number) {
    int sum = 0;
    for (int i = 1; i <= number ~/ 2; i++) {
      if (number % i == 0) {
        sum += i;
      }
    }
    return sum;
  }
}
